docs/source/examples/spot-jobs.rst

```diff
@@ -275,26 +275,34 @@ you can still tear it down manually with
 Customizing spot controller resources
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-You may customize the resources of the spot controller for the following reasons:
+You may want to customize the resources of the spot controller for several reasons:
 
-1. Enforcing the spot controller to run on a specific location. (Default: cheapest location)
-2. Changing the maximum number of spot jobs that can be run concurrently. (Default: 16)
-3. Changing the disk_size of the spot controller to store more logs. (Default: 50GB)
+1. Use a lower-cost controller (if you have a low number of concurrent spot jobs).
+2. Enforcing the spot controller to run on a specific location. (Default: cheapest location)
+3. Changing the maximum number of spot jobs that can be run concurrently, which is 2x the vCPUs of the controller. (Default: 16)
+4. Changing the disk_size of the spot controller to store more logs. (Default: 50GB)
 
 To achieve the above, you can specify custom configs in :code:`~/.sky/config.yaml` with the following fields:
 
 .. code-block:: yaml
 
   spot:
+    # NOTE: these settings only take effect for a new spot controller, not if
+    # you have an existing one.
     controller:
       resources:
-        # All configs below are optional
-        # 1. Specify the location of the spot controller.
+        # All configs below are optional.
+        # Specify the location of the spot controller.
         cloud: gcp
         region: us-central1
-        # 2. Specify the maximum number of spot jobs that can be run concurrently.
+        # Specify the maximum number of spot jobs that can be run concurrently.
         cpus: 4+  # number of vCPUs, max concurrent spot jobs = 2 * cpus
-        # 3. Specify the disk_size of the spot controller.
+        # Specify the disk_size in GB of the spot controller.
         disk_size: 100
 
 The :code:`resources` field has the same spec as a normal SkyPilot job; see `here <https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html>`__.
+
+.. note::
+  These settings will not take effect if you have an existing controller (either
```

Comment:
Shall we add a note to say that changing the config but not terminate the spot controller could lead to a failure?
Related: #2703 (for context of this PR, I'm waiting for Zhanghao's opinion)

Reply:
Probably should add such a warning as part of #2703, since right now it doesn't error out. Wdyt?

Reply:
Now it still errors out due to `sky.exceptions.ResourcesMismatchError`, though arguably the error message is informative, so it should be fine to leave to #2703 ; )

```bash
sky spot launch whoami
Task from command: whoami
Managed spot job 'sky-cmd' will be launched on (estimated):
I 11-05 10:56:23 optimizer.py:674] == Optimizer ==
I 11-05 10:56:23 optimizer.py:686] Target: minimizing cost
I 11-05 10:56:23 optimizer.py:697] Estimated cost: $0.0 / hour
I 11-05 10:56:23 optimizer.py:697] 
I 11-05 10:56:23 optimizer.py:770] Considered resources (1 node):
I 11-05 10:56:23 optimizer.py:818] ----------------------------------------------------------------------------------------------------------------
I 11-05 10:56:23 optimizer.py:818]  CLOUD   INSTANCE              vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE                 COST ($)   CHOSEN   
I 11-05 10:56:23 optimizer.py:818] ----------------------------------------------------------------------------------------------------------------
I 11-05 10:56:23 optimizer.py:818]  GCP     n2-standard-8[Spot]   8       32        -              northamerica-northeast2-a   0.04          ✔     
I 11-05 10:56:23 optimizer.py:818]  AWS     m6i.2xlarge[Spot]     8       32        -              eu-north-1c                 0.12                
I 11-05 10:56:23 optimizer.py:818] ----------------------------------------------------------------------------------------------------------------
I 11-05 10:56:23 optimizer.py:818] 
Launching the spot job 'sky-cmd'. Proceed? [Y/n]: 
Launching managed spot job 'sky-cmd' from spot controller...
Launching spot controller...

sky.exceptions.ResourcesMismatchError: Requested resources do not match the existing cluster.
  Requested:    1x AWS(cpus=4, disk_size=50) 
  Existing:     1x GCP(n2-standard-4, disk_size=50)
To fix: specify a new cluster name, or down the existing cluster first: sky down sky-spot-controller-402b1bba
```

Reply:
Yep let's leave it there. I think the situation that same cloud but smaller controller is more common.